modules/gdnative/gdnative_builders.py

```diff
@@ -128,7 +128,7 @@ def generate_core_extension_struct(core):
     return "\n".join(out)
 
 
-def _build_gdnative_api_struct_source(api):
+def _build_gdnative_api_struct_source(api, env):
```

Comment:
I'm not sure I like the idea to make the GDNative API (and thus ABI?) conditional. Might not be worth the (likely quite small) gain from excluding a handful of definitions.